### PR TITLE
add option to set custom package names

### PR DESCRIPTION
**Pull request by bastelfreak**

### CODE REVIEW

File: manifests/install.pp [hunk 1]
```diff
@@ -1,11 +1,14 @@
 # Install TFTP
-class tftp::install {
-  package { $::tftp::params::package:
+class tftp::install (
+  $package          = $tftp::params::package,
```
Comment (domcleal) [inline manifests/install.pp]: Shouldn't the default value be `$::tftp::package`, not params, if the override from the main tftp class is to work?

File: manifests/init.pp [hunk 2]
```diff
@@ -18,8 +20,18 @@
 #    root => '/tftpboot',
 #  }
 #
+# * Configure a TFTP server with non-default package name:
+#  class { 'tftp:'
+#    package => 'tftp-hpa-destruct',
+#  }
+#
+#
+#
```
Comment (domcleal) [inline manifests/init.pp]: Lots of superfluous lines here.

File: manifests/init.pp [hunk 3]
```diff
@@ -5,6 +5,8 @@
 # === Parameters
 #
 # $root:: Configures the root directory for the TFTP server
+# $package:: name of the tftp package
+# $syslinux_package:: essential for pxe boot
```
Comment (domcleal) [inline manifests/init.pp]: A more useful description would be good, e.g. the name of the package providing SYSLINUX

**Comment by domcleal** [comment]
There are a handful of specs too in this module, if you could add a test to show the parameters work, that would be good.  It may verify my first comment about parameter defaults.

**Comment by bastelfreak** [comment]
rebased. Added tests, removed duplicated lines, fixed description.

File: manifests/install.pp [hunk 4]
```diff
@@ -1,11 +1,14 @@
 # Install TFTP
-class tftp::install {
-  package { $::tftp::params::package:
+class tftp::install (
+  $package          = $::tftp::package,
+  $syslinux_package = $::tftp::syslinux_package,
+) inherits tftp::params {
```
Comment (domcleal) [inline manifests/install.pp]: The inherits is probably not required now, it can be removed.

File: spec/classes/init_spec.rb [hunk 5]
```diff
@@ -125,6 +125,28 @@
...
+
+        it 'should install custom tftp package' do
+          should contain_package('tftp-hpa-destruct').with({
+            :ensure => 'installed',
+            :alias  => 'tftp-server',
+          })
+        end
+      end
+
+      context 'with custom syslinix package set to my-own-syslinux' do
```
Comment (domcleal) [inline spec/classes/init_spec.rb]: typo, syslinux

File: spec/classes/init_spec.rb [hunk 6]
```diff
@@ -125,6 +125,28 @@
           # not supported
         end
       end
+
+      context 'with custom tftp package set to tftp-hpa-destruct' do
+        let :params do {
+            :package => 'tftp-hpa-destruct',
```
Comment (domcleal) [inline spec/classes/init_spec.rb]: Nitpick, two-space indentation (and also on line 144, the other :params block).

**Comment by domcleal** [comment]
Thanks, a couple of minor comments but otherwise looks good. :+1:

**Comment by bastelfreak** [comment]
@domcleal should be fine now.

**Comment by ekohl** [comment]
:+1:

**Comment by mmoll** [comment]
Merged, Danke @bastelfreak!

**Pull request merged**